### PR TITLE
Add support for signet genesis block

### PR DESCRIPTION
**Pull request by remyers**
See issue #60 and Eclair discussion/request [#2382](https://github.com/ACINQ/eclair/discussions/2382)

This PR also adds check of the final genesis block hashes.

### CODE REVIEW

File: src/commonMain/kotlin/fr/acinq/bitcoin/Block.kt [hunk 1]
```diff
@@ -373,5 +373,13 @@ public data class Block(@JvmField val header: BlockHeader, @JvmField val tx: Lis
             )
         )
 
+        @JvmField
+        public val SignetGenesisBlock: Block = LivenetGenesisBlock.copy(
```
Comment (t-bast) [inline src/commonMain/kotlin/fr/acinq/bitcoin/Block.kt:377]: Do you know what the `Segnet` one above refers to?

**Review by t-bast — COMMENTED** [review]
LGTM, just wondering what the `Segnet` one we have is supposed to be...

File: src/commonMain/kotlin/fr/acinq/bitcoin/Block.kt (hunk 1 above)
Comment (sstone) [reply src/commonMain/kotlin/fr/acinq/bitcoin/Block.kt:377]: `Segnet` was a test network that was used to test segwit in 2016. It's most likely dead by now.

Comment (t-bast) [reply src/commonMain/kotlin/fr/acinq/bitcoin/Block.kt:377]: Interesting, I guess we can remove it in this PR then? It's a bit confusing with `Signet`...

Comment (sstone) [reply src/commonMain/kotlin/fr/acinq/bitcoin/Block.kt:377]: Agreed, it was ported from `bitcoin-lib` but is useless now.

**Comment by t-bast** [comment]
@remyers can you remove the `Segnet` hash in this PR as well (and in https://github.com/ACINQ/bitcoin-lib/pull/70)?

**Comment by remyers** [comment]
> @remyers can you remove the `Segnet` hash in this PR as well (and in [ACINQ/bitcoin-lib#70](https://github.com/ACINQ/bitcoin-lib/pull/70))?

It looks like I will have to recreate on `testnet` some tests in [SegwitTestsCommon.kt](https://github.com/ACINQ/bitcoin-kmp/blob/e69e17c236359d6e164eb5c9ea3837f632be86af/src/commonTest/kotlin/fr/acinq/bitcoin/SegwitTestsCommon.kt#L117) that reference `segnet` transactions.

**Comment by t-bast** [comment]
> It looks like I will have to recreate on testnet some tests in [SegwitTestsCommon.kt](https://github.com/ACINQ/bitcoin-kmp/blob/e69e17c236359d6e164eb5c9ea3837f632be86af/src/commonTest/kotlin/fr/acinq/bitcoin/SegwitTestsCommon.kt#L117) that reference segnet transactions.

Good point, but I think most of these tests are obsolete, and were only useful before segwit was activated. You can probably greatly simplify this test file by just verifying that p2wpkh and p2wsh work as expected, without referencing segnet transactions, this will remove some technical debt!

**Comment by remyers** [comment]
The tests seemed useful, so I just ported them to testnet in [2581f47](https://github.com/ACINQ/bitcoin-kmp/pull/61/commits/2581f47815f647aa1b7fa1f8ad8e27719efba2f1) after removing segnet.

**Review by t-bast — APPROVED** [review]
LGTM :100: 

segwit

Yay saved fees!

**Pull request merged**